engine/src/mac-dialog.mm

```diff
@@ -605,70 +563,82 @@ - (BOOL)panel:(id)sender shouldShowFilename:(NSString *)filename
 
 ////////////////////////////////////////////////////////////////////////////////
 
-void MCPlatformBeginFileDialog(MCPlatformFileDialogKind p_kind, MCPlatformWindowRef p_owner, MCStringRef p_title, MCStringRef p_prompt, MCStringRef *p_types, uint4 p_type_count, MCStringRef p_initial)
+void MCPlatformBeginFolderOrFileDialog(MCPlatformFileDialogKind p_kind, MCPlatformWindowRef p_owner, MCStringRef p_title, MCStringRef p_prompt, MCStringRef *p_types, uint4 p_type_count, MCStringRef p_initial, bool p_is_file)
```

Comment:
Does this need the 'p_is_file' parameter, since the p_kind parameter can already indicate a folder dialog?